src/Services/InventoryUpdateService.php

```diff
@@ -37,319 +54,524 @@ class InventoryUpdateService
+
+    /** @var ExternalLogs */
     $externalLogs = pluginApp(ExternalLogs::class);
-    /** @var VariationSearchRepositoryContract $variationSearchRepository */
-    $variationSearchRepository = pluginApp(VariationSearchRepositoryContract::class);
-    /** @var AbstractConfigHelper $configHelper */
-    $configHelper = pluginApp(AbstractConfigHelper::class);
-
-    // look up item mapping method at this level to ensure consistency and improve efficiency
-    $itemMappingMethod = $configHelper->getItemMappingMethod();
-
-    $loggerContract->debug(
-      TranslationHelper::getLoggerKey(self::LOG_KEY_INVENTORY_UPDATE_START),
-      [
-        'additionalInfo' => ['fullInventory' => (string) $fullInventory],
-        'method' => __METHOD__
-      ]
-    );
+    try {
 
-    $page = 0;
-    $inventorySaveTotal = 0;
-    $inventorySaveSuccess = 0;
-    $inventorySaveFail = 0;
-    $saveInventoryDuration = 0;
-    $savedInventoryDuration = 0;
+      if ($this->syncIsBlocked($fullInventory)) {
+        throw new InventorySyncBlockedException("Another inventory sync is in progress, preventing this one from starting");
+      }
 
-    $errorMessage = null;
-    $amtOfDtosForPage = 0;
+      if (!$fullInventory) {
+        $lastGoodFullSyncStart = $this->statusService->getLastSuccessfulAttemptTime(true);
+        if (!isset($lastGoodFullSyncStart) || empty($lastGoodFullSyncStart)) {
+          $this->logger->info(
+            TranslationHelper::getLoggerKey(self::LOG_KEY_NO_SYNCS),
+            [
+              'additionalInfo' => [],
+              'method' => __METHOD__
+            ]
+          );
 
-    try {
-      $fields = $this->getResultFields();
-      /* Page size is tuned for a balance between memory usage (in plentymarkets) and number of transactions  */
-      $fields['itemsPerPage'] = AbstractConfigHelper::INVENTORY_ITEMS_PER_PAGE;
-      $variationSearchRepository->setFilters($this->getFilters($fullInventory));
+          // NOTICE: due to recursion, this log will show up AFTER the logs for the Full Sync!
+          $externalLogs->addInfoLog("Forced a full inventory sync as there are no successful full syncs on record");
+
+          return $this->sync(true, $manual);
+        }
+      }
+
+      // look up item mapping method at this level to ensure consistency and improve efficiency
+      $itemMappingMethod = $this->configHelper->getItemMappingMethod();
+
+      // TODO: remove dependency on VariationSearchRepositoryContract by replacing with a Wayfair wrapper
+      /** @var VariationSearchRepositoryContract */
+      $variationSearchRepository = pluginApp(VariationSearchRepositoryContract::class);
 
       // stock buffer should be the same across all pages of inventory
-      $stockBuffer = self::getNormalizedStockBuffer($configHelper, $loggerContract);
+      $stockBuffer = $this->getNormalizedStockBuffer();
+
+      $searchParameters = $this->getDefaultSearchParameters();
+
+      $pageNumber = 0;
```

Comment:
Per this comment should we actually be starting at 1? https://forum.plentymarkets.com/t/what-is-the-proper-starting-page-number/595947/3

Reply:
I never tested if `1` works. We know `0` gets us results which is why we still use `0` here, but yeah I should test if `1` works.

Reply:
That fact that 0 does return results makes me wary of switching to 1 as we could end up missing data. It's worth testing and seeing the difference in variations being synced

Reply:
yeah I tested `1` in another branch, and we got the same amount of Variations as with `0`, so I'll cautiously say that `1` is an appropriate starting number.